dataRetriever/resolvers/trieNodeResolver.go

```diff
@@ -115,14 +115,14 @@ func (tnRes *TrieNodeResolver) resolveOnlyRequestedHashes(hashes [][]byte, nodes
 			continue
 		}
 
+		spaceUsed += len(serializedNode)
```

Comment:
yes, otherwise this function can not get the single large element. This is mandatory as found out in the integration test and in the first implementation.